packages/store/src/StoreCore.sol

```diff
@@ -291,6 +291,39 @@ library StoreCore {
     }
   }
 
+  function popFromField(bytes32 tableId, bytes32[] memory key, uint8 schemaIndex, uint256 byteLengthToPop) internal {
+    Schema schema = getSchema(tableId);
+
+    if (schemaIndex < schema.numStaticFields()) {
+      revert IErrors.StoreCore_NotDynamicField();
+    }
+
+    // TODO add pop-specific event and hook to avoid the storage read? (https://github.com/latticexyz/mud/issues/444)
```

Comment:
before we start promoting "MUD the protocol", we should probably figure out all the events we're gonna want, because it sounds dangerous to add something like a push event down the line where some old clients may not properly support it?

Reply:
though I guess we could still have a backwards compatibility layer or hook that does the full read + `emit StoreSetField` for supporting old clients

Reply:
Agree we should add these events before the first stable release